x/crosschain/keeper/keeper.go

```diff
@@ -8,14 +8,22 @@ import (
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
+	"github.com/zeta-chain/zetacore/pkg/chains"
 	"github.com/zeta-chain/zetacore/x/crosschain/types"
 )
 
+type CCTXGateway interface {
```

Comment:
Need comment explaining the idea behind this interface.
We can reuse the Notion doc description for this